application/server/server.go

```diff
@@ -490,23 +490,23 @@ func startOfflineDetection(c *config.Config) { //nolint:unused // this is gonna
 	}()
 }
 
-func deleteExpiredCache() {
-	w := workspace.Get()
+func deleteExpiredCache(c *config.Config) {
+	w := c.Workspace()
 	var folderList []string
 	for _, f := range w.Folders() {
 		folderList = append(folderList, f.Path())
 	}
-	di.ScanPersister().Clear(folderList, true)
+	c.Workspace().GetScanSnapshotClearerExister().Clear(folderList, true)
 }
 
-func periodicallyCheckForExpiredCache() {
+func periodicallyCheckForExpiredCache(c *config.Config) {
 	for {
-		deleteExpiredCache()
+		deleteExpiredCache(c)
 		time.Sleep(time.Duration(persistence.ExpirationInSeconds) * time.Second)
 	}
 }
 
-func addWorkspaceFolders(c *config.Config, params types.InitializeParams, w *workspace.Workspace) {
+func addWorkspaceFolders(c *config.Config, params types.InitializeParams, w types.Workspace) {
 	const method = "addWorkspaceFolders"
```

Comment:
`w types.Workspace` argument isn't needed anymore here we can get it from config

Reply:
true